cmd/check/main.go

```diff
@@ -35,6 +42,68 @@ func main() {
 		return
 	}
 
+	if req.Source.Ecr {
+		if req.Source.AwsAccessKeyId != "" {
+			os.Setenv("AWS_ACCESS_KEY_ID", req.Source.AwsAccessKeyId)
+		}
+		if req.Source.AwsSecretAccessKey != "" {
+			os.Setenv("AWS_SECRET_ACCESS_KEY", req.Source.AwsSecretAccessKey)
+		}
+		if req.Source.AwsRegion != "" {
+			os.Setenv("AWS_REGION", req.Source.AwsRegion)
+		}
+		mySession := session.Must(session.NewSession())
+		client := ecr.New(mySession)
+		// If a role arn has been supplied, then assume role and get a new session
+		if req.Source.AwsRoleArn != "" {
+			creds := stscreds.NewCredentials(mySession, req.Source.AwsRoleArn)
+			client = ecr.New(mySession, &aws.Config{Credentials: creds})
+		}
+		input := &ecr.GetAuthorizationTokenInput{}
+		result, err := client.GetAuthorizationToken(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
```

Comment:
Can we just do `logrus.Errorf("failed to get authorization token: %s", err)` and bail? The type checking and switching seems unnecessary since we're just gonna log and fail anyway.

note: if we pull this all into a method on `req.Source` it could probably just return the error, and the logging + exiting (which is missing btw) could be done in the `cmd/*/main.go` instead